common/custom_claims.c

```diff
@@ -0,0 +1,219 @@
+// Copyright (c) Open Enclave SDK contributors.
+// Licensed under the MIT License.
+
+#include <openenclave/attestation/custom_claims.h>
+#include <openenclave/internal/raise.h>
+#include <openenclave/internal/safemath.h>
+#include "oe_host_stdlib.h"
+
+static void _free_claim(oe_claim_t* claim)
+{
+    oe_free(claim->name);
+    oe_free(claim->value);
+}
+
+static oe_result_t _free_claims(oe_claim_t* claims, size_t claims_length)
+{
+    if (!claims)
+        return OE_OK;
+
+    for (size_t i = 0; i < claims_length; i++)
+        _free_claim(&claims[i]);
+    oe_free(claims);
+    return OE_OK;
+}
+
+static oe_result_t _add_claim(
+    oe_claim_t* claim,
+    const void* name,
+    size_t name_size, // Must cover the '\0' at end of string
```

Comment:
Does it need to be a string?  if so (and line 155 implies it has to be), then why isn't name a const char* and delete the name_size arg?
If it does not need to be a string (and I think there's a good argument for why requiring a string would be problematic for constrained devices, but maybe the answer for them is don't use these helper functions, write your own), then the comment is misleading and oe_claim_t would need to have a name_size field.  Point is that the type of the name argument here, and the type of the name field in oe_claim_t should be the same.  Currently they're different.

Reply:
This is called from the deserialize function, we pass the name, name_size, value and value_size and then fill the oe_claim_t array ( line 36, 41)

Reply:
Ok, I guess since this encoding is already quite inefficient as noted above, having it use an extra null terminator in the buffer is ok.  Anyone worried about the overhead would never use these helper functions and would instead write their own more efficient ones I guess.

And since the name pointer is into the serialized buffer, it should probably be of type "const uint8_t*" then.